etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-legacy.R

```diff
@@ -0,0 +1,264 @@
+# Can't be read as a FWF like the ones below because some rows have extra chars
+cc_dli_senfrr <- map_dfr(files_cc_dli_senfrr$Key, \(f) {
+  aws.s3::s3read_using(
+    object = f,
+    bucket = AWS_S3_RAW_BUCKET,
+    FUN = readr::read_delim,
+    delim = "|",
+    trim_ws = TRUE,
+    col_names = c(
+      "pin", "year", "tax_year", "tax_type", "rec_code",
+      "base_value_year_manual_calculation_ind", "base_value_year",
+      "base_value_year_total_eav", "cur_year_total_eav", "cur_year_final_eav",
+      "birth_date", "applicant_old_name", "applicant_address", "applicant_city",
+      "applicant_state", "applicant_zip", "applicant_status",
+      "first_app_received_date", "qualified_date",
+      "homeowner_base_year_eq_factor", "bldg_units", "building_shares",
+      "base_value_year_class", "num_units_w_homeowner_exemption",
+      "num_units_w_homestead_exemption", "num_shares_with_senior_freeze",
+      "coop_senior_shares", "pct_senior_shares",
+      "num_shares", "pct_of_shares", "sf_percent"
+    ),
+    col_types = cols(
+      pin = col_character(),
+      year = col_character(),
+      tax_year = col_character(),
+      tax_type = col_character(),
+      rec_code = col_character(),
+      base_value_year_manual_calculation_ind = col_character(),
+      base_value_year = col_character(),
+      base_value_year_total_eav = col_integer(),
+      cur_year_total_eav = col_integer(),
+      cur_year_final_eav = col_integer(),
+      birth_date = col_character(),
+      applicant_old_name = col_character(),
+      applicant_address = col_character(),
+      applicant_city = col_character(),
+      applicant_state = col_character(),
+      applicant_zip = col_character(),
+      applicant_status = col_character(),
+      first_app_received_date = col_character(),
+      qualified_date = col_character(),
+      homeowner_base_year_eq_factor = col_double(),
+      bldg_units = col_integer(),
+      building_shares = col_integer(),
+      base_value_year_class = col_character(),
+      num_units_w_homeowner_exemption = col_integer(),
+      num_units_w_homestead_exemption = col_integer(),
+      num_shares_with_senior_freeze = col_integer(),
+      coop_senior_shares = col_integer(),
+      pct_senior_shares = col_integer(),
+      num_shares = col_integer(),
+      pct_of_shares = col_integer(),
+      sf_percent = col_integer()
+    )
+  ) %>%
+    mutate(
+      across(
+        c(year, tax_year),
+        \(x) ifelse(substr(x, 1, 1) == "9", paste0("19", x), paste0("20", x))
+      ),
+      across(starts_with("applicant_"), \(x) str_trim(str_squish(x))),
+      applicant_zip = str_sub(applicant_zip, 1, 5),
+      birth_date = lubridate::mdy(birth_date),
+      first_app_received_date = lubridate::ymd(first_app_received_date),
+      base_value_year_class = str_sub(base_value_year_class, 4, 6),
+      qualified_date = lubridate::ymd(qualified_date),
+      source_file = {{ file }}
```

Comment:
**[Question, non-blocking]** R newbie question: Based on the [dplyr docs](https://dplyr.tidyverse.org/articles/programming.html#indirection), I think I understand that we're embracing the `file` symbol here because we want to use an env-variable value to reference a data-variable. Is that right? If so, where is `file` defined in the environment?

Reply:
Whoops, I'm dumb and didn't change `file` to `f` after I changed the function arg name. Fixed in 9f6f41b.

The double curly braces actually aren't necessary in this case since we want the `f` arg to evaluate to a string (instead of a [quosure](https://rlang.r-lib.org/reference/topic-quosure.html#foreign-and-local-expressions)), but it also doesn't hurt. One benefit of the double-curly braces here is that we could have a function argument with the same name as an existing column but still use it in assignment. For example:

```r
df <- tibble(a = 1, b = 2)

test <- function(data, a) {
  data %>%
    mutate(a = a)
}
test(df, a = 3)
```

Would return a tibble where `a = 1`, while the below would return a tibble where `a = 3`:

```r
df <- tibble(a = 1, b = 2)

test <- function(data, a) {
  data %>%
    mutate(a = {{ a }})
}
test(df, a = 3)
```

Reply:
Very clear explanation, thanks!